ports/libtorch/portfile.cmake

```
@@ -0,0 +1,180 @@
+    list(APPEND FEATURE_OPTIONS -DUSE_NATIVE_ARCH=ON)
+endif()
+if("dist" IN_LIST FEATURES)
+    if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
+        list(APPEND FEATURE_OPTIONS -DUSE_TENSORPIPE=ON)
+    endif()
+    if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_OSX)
+        list(APPEND FEATURE_OPTIONS -DUSE_LIBUV=ON)
+    endif()
+endif()
+
+if(VCPKG_TARGET_IS_LINUX)
+    # Linux package `libnuma-dev`
+    find_library(Numa_LIBPATH NAMES numa PATHS "/usr/lib" "/usr/lib/x86_64-linux-gnu")
+    if(Numa_LIBPATH)
+        message(STATUS "Detected numa: ${Numa_LIBPATH}")
+        list(APPEND FEATURE_OPTIONS -DUSE_NUMA=ON)
+    else()
+        message(STATUS "To enable USE_NUMA build option, install 'libnuma-dev' package")
+        list(APPEND FEATURE_OPTIONS -DUSE_NUMA=OFF)
+    endif()
+else()
+    list(APPEND FEATURE_OPTIONS -DUSE_NUMA=OFF)
+endif()
+
+if(VCPKG_TARGET_IS_OSX)
+    list(APPEND FEATURE_OPTIONS -DBLAS=Accelerate) # Accelerate.framework will be used for Apple platforms
+else()
+    list(APPEND FEATURE_OPTIONS -DBLAS=Eigen)
+endif()
+
+if("tbb" IN_LIST FEATURES)
+    list(APPEND FEATURE_OPTIONS
+        -DMKLDNN_CPU_RUNTIME=TBB
+    )
+endif()
+
+if(VCPKG_TARGET_IS_ANDROID)
+    list(APPEND FEATURE_OPTIONS
+        -DINTERN_BUILD_MOBILE=ON
+        -DBUILD_JNI=ON
+        -DUSE_NNAPI=ON
+    )
+else()
+    list(APPEND FEATURE_OPTIONS -DINTERN_BUILD_MOBILE=OFF)
+endif()
+
+string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "static" USE_STATIC_RUNTIME)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        ${FEATURE_OPTIONS}
+        -DPython3_EXECUTABLE="${PYTHON3}"
+        -DCAFFE2_USE_MSVC_STATIC_RUNTIME=${USE_STATIC_RUNTIME}
+        -DBUILD_CUSTOM_PROTOBUF=OFF -DUSE_LITE_PROTO=OFF
+        -DBUILD_TEST=OFF -DATEN_NO_TEST=ON
+        -DUSE_SYSTEM_LIBS=ON
+        -DBUILD_PYTHON=OFF
+        -DUSE_GLOO=${VCPKG_TARGET_IS_LINUX}
+        -DUSE_MPI=${VCPKG_TARGET_IS_LINUX} # Linux package `libopenmpi-dev`
```

Do we need to have this installed in our VMs? Need to confirm what happens if this package is missing. (If missing it needs to fail to build not silently do nothing to preserve build-path-independence)

---

It looks like `vcpkg-dependencies.cmake` says `REQUIRED`.

---

The internal log is too complicated. I think we can `OFF` it if VM needs too much changes.

---

Let me check it's `REQUIRED` once more

---

`mpi` port ? Also this should probably be a feature instead.